dockerfiles/centos-7-pg11/Dockerfile

```
@@ -38,7 +38,8 @@ RUN if [[ centos   == oraclelinux ]] && [[ 7 == 7 ]]; then yum install -y wget \
 
 
 # install build tools and PostgreSQL development files
-RUN ( yum install -y https://download.postgresql.org/pub/repos/yum/reporpms/EL-7-x86_64/pgdg-redhat-repo-latest.noarch.rpm \
+RUN yum update -y \
+    && ( yum install -y https://download.postgresql.org/pub/repos/yum/reporpms/EL-7-x86_64/pgdg-redhat-repo-latest.noarch.rpm \
```

Normally, we would run that command as in below suggestion, since the pr title claims that it only addresses to ol/7:
```suggestion
RUN if [[ centos == oraclelinux ]] && [[ 7 == 7 ]]; then yum update -y

RUN ( yum install -y https://download.postgresql.org/pub/repos/yum/reporpms/EL-7-x86_64/pgdg-redhat-repo-latest.noarch.rpm \
```

However, I guess there is no harm doing that for all rpm's, right ?

---

### I thinks there is a potential for all of the packages since new repo is being added. I changed the order of the yum update -y to cover the two repos  being added below

---

I have several remarks:
1. `yum` does not really need explicit commands to update the cache.
2. If you wish to install the latest information on all available packages on a repo, you should run `yum check-update` instead, and make sure you consume the return value properly
3. `yum update` is similar to running `apt update; apt upgrade` as it installs all updates to currently installed packages.
4. it does not really make sense to me to update packages for ol7 and not do that on other distros.

After you go over these comments. can you update the PR and leave a comment on why you want these changes? Please focus on why, and skip what or how as your code changes should speak for itself.

---

I agree with all, thanks Hanefi

---

I already focused on why in my first comment  '''I thinks there is a potential for all of the packages since new repo is being added.''' I think this is a good why. I 'm checking your comment about yum update by which you gave the answer for the why. Thanks for the input